modules/claims_api/app/controllers/claims_api/v2/veterans/disability_compensation_controller.rb

```diff
@@ -22,7 +22,11 @@ class DisabilityCompensationController < ClaimsApi::V2::Veterans::Base
         FORM_NUMBER = '526'
 
         skip_before_action :validate_json_format, only: [:attachments]
-        before_action :shared_validation, :file_number_check, only: %i[submit validate]
+        # Custom validations for 526 submission, we must check this first
+        before_action :custom_validation, :file_number_check, only: %i[submit validate]
+        before_action :validate_json_schema, only: %i[submit validate generate_pdf]
+        # If all we have are custom validation errors we raise them here, after validating JSON above
+        before_action :check_for_custom_validation_errors, only: %i[submit validate]
```

Comment:
Should we remove line 29? I am not sure I understand why we have line 26 and 29.

Reply:
Basically I am trying to break this sequence out into before actions to avoid any duplication because we need to validate the JSON for the generatePDF endpoint but not the custom validation

so in the old method we had
```
        def shared_validation
          # Custom validations for 526 submission, we must check this first
          @disability_compensation_validation_errors = validate_form_526_submission_values!(target_veteran)
          # JSON validations for 526 submission, will combine with previously captured errors and raise
          validate_json_schema
          # if we get here there were only validations file errors
          if @disability_compensation_validation_errors
            raise ::ClaimsApi::Common::Exceptions::Lighthouse::JsonDisabilityCompensationValidationError,
                  @disability_compensation_validation_errors
          end
        end
```

So the way this currently runs in this method is we run our custom validations and collect any errors.  
```
@disability_compensation_validation_errors = validate_form_526_submission_values!(target_veteran)
```
We then run the JSON validator and any errors collected will be added.  _And_, if the JOSN validator does find any errors it will also raise at that point. 
```
validate_json_schema
```
 If no errors are collected by the JSON validator, there is no raise occurs yet, so the next step checks and raises anything that was previously collected by the custom validations file.
```
          if @disability_compensation_validation_errors
            raise ::ClaimsApi::Common::Exceptions::Lighthouse::JsonDisabilityCompensationValidationError,
                  @disability_compensation_validation_errors
          end
```
Which is why line 29 exists above.

If line 26 finds errors but line 27 does not, then line 29 will make sure those errors are still raised.

If line 26 finds errors and line 27 finds errors, line 27 raises them (which cannot be avoided from what I found, which is why the JOSN validator had to be moved to run second in line when returning a full error array).

As I mentioned in slack.  We could revert these, and then just add a before action to validate the JSON only form the generatePDF endpoint if we think that is better.  Also open to other suggestions of course.